tests/tracer/telemetry/test_writer.py

```diff
@@ -243,6 +243,12 @@ def test_send_request_exception(telemetry_writer):
         )
 
 
+def test_telemetry_graceful_shutdown(mock_time, mock_send_request, telemetry_writer):
+    telemetry_writer.start()
+    telemetry_writer.stop()
+    assert httpretty.last_request().parsed_body == _get_request_body({}, "app-closing")
```

Comment:
should we have test that the atexit hook does the proper thing?

Reply:
I created a follow up issue to validate telemetry payloads in the test agent.